Tools/TestWebKitAPI/Tests/WebKit/mac/AttributedSubstringForProposedRange.mm

```diff
@@ -82,4 +82,32 @@ static void invalidMessageFunction(WKStringRef messageName)
     EXPECT_EQ(didReceiveInvalidMessage, false);
 }
 
+TEST(WebKit, AttributedSubstringForProposedRangeWithAnchorWithInvalidHref)
+{
+    WKRetainPtr context = adoptWK(Util::createContextWithInjectedBundle());
+    PlatformWebView webView(context.get());
+
+    WKPageNavigationClientV0 loaderClient;
+    zeroBytes(loaderClient);
+    loaderClient.base.version = 0;
+    loaderClient.didFinishNavigation = didFinishNavigation;
+    loaderClient.webProcessDidCrash = processDidCrash;
+    WKPageSetPageNavigationClient(webView.page(), &loaderClient.base);
+
+    WKContextSetInvalidMessageFunction(invalidMessageFunction);
+
+    WKRetainPtr url = adoptWK(Util::createURLForResource("anchor-element-with-invalid-href", "html"));
+    WKPageLoadURL(webView.page(), url.get());
+    Util::run(&didFinishLoad);
+
+    [webView.platformView() attributedSubstringForProposedRange:NSMakeRange(0, 3) completionHandler: ^(NSAttributedString *attrString, NSRange actualRange) {
+        didFinishTest = true;
```

Comment:
Can we check the value of the NSAttributedString here?

Reply:
Sure could, though I don't think we care about the contents too much? (i.e. what about it would we want to check? The actual string/length?)